### PR TITLE
Add support for service suspensions and unsuspensions

### PR DESCRIPTION
**Pull request by abdyfranco**

### CODE REVIEW

File: apis/teamspeak_api.php [hunk 1]
```diff
@@ -370,6 +370,78 @@ public function stopServer($sid)
...
+    public function unsuspendServer($sid)
+    {
+        try {
+            // Build the parameters array
+            $api_params = [
+                'virtualserver_autostart' => 1
+            ];
+
+            // Fetch the virtual server instance, and modify it
+            $instance = $this->apiRequest('serverquery')->serverList()[$sid];
```
Comment (JReissmueller) [inline apis/teamspeak_api.php]: I'd prefer we use $this->getServer($sid);

File: apis/teamspeak_api.php [hunk 2]
```diff
@@ -370,6 +370,78 @@ public function stopServer($sid)
...
+    public function suspendServer($sid)
+    {
+        try {
+            // Build the parameters array
+            $api_params = [
+                'virtualserver_autostart' => 0
+            ];
+
+            // Fetch the virtual server instance, and modify it
+            $instance = $this->apiRequest('serverquery')->serverList()[$sid];
```
Comment (JReissmueller) [inline apis/teamspeak_api.php]: I'd prefer we use $this->getServer($sid);

Comment (abdyfranco) [reply apis/teamspeak_api.php]: Done.

File: apis/teamspeak_api.php [hunk 3]
```diff
@@ -370,6 +370,78 @@ public function stopServer($sid)
...
+    {
+        try {
+            // Build the parameters array
+            $api_params = [
+                'virtualserver_autostart' => 1
+            ];
+
+            // Fetch the virtual server instance, and modify it
+            $server = $this->getServer($sid);
+            $result = $server->modify($api_params);
```
Comment (JReissmueller) [inline apis/teamspeak_api.php]: So a couple things here.  First, the result of getServer() is an stdClass object and has no modify() method.  You are looking for $server->instance->modify(), but also getServer() may return an object with errors and no ->instance property so you should check for that and probably return that object if errors exist.  These same things of course also apply to the suspendServer() method.

One other thing I encountered is that the API did not allow me to modify the server while it was stopped, but only if I ran startServer() before this modify() call.

**Pull request merged**